testing/test_main.py

```diff
@@ -201,6 +201,12 @@ def test_does_not_exist(self, invocation_path: Path) -> None:
         ):
             resolve_collection_argument(invocation_path, "foobar")
 
+        with pytest.raises(
```

Comment:
This test doesn't look right to me -- the path here should be `src/pkg/test.py` not `src/pkg/test.py[a::b]`.

Reply:
but considering that pytest supports non python tests. In such case someone could have a file with virtually any name E.g. `test.py[a::b]` 

Does this mean that `[` and `::` should be avoided in test file names in such case ?

Reply:
In such case having `rest` and `not parts` should be invalid right ? I mean should `resolve_argument` raise an exception at such condition ?
E.g.

```py
# invalid usage for the split :: operator
resolve_collection_argument(Path(), "test_foo.py[a::b]")

# rest = 'a::b]'
# parts = []
```

Reply:
Right now, `test_foo.py[a::b]` ends up just ignoring the `[a::b]` part. I agree it's not so great.

Note that 2442034a1eca53dae4b3141732660f3384b6d159 explicitly changed the split order of `::` and `[`.

IMO, the should be the behavior (I'm always assuming the returned `Path` actually exists in the filesystem):

1. `test_foo.py[a::b]` -> `(Path('test_foo.py[a::b]'), [])`.
2. `test_foo.py::` -> `(Path('test_foo.py'), [])` (special case for empty string)
3. `test_foo.py::test_it[a::b]` -> `(Path('test_foo.py'), 'test_it[a::b]')`

This behavior requires `resolve_collection_argument` parsing to be a bit more elaborate, just splitting by `::` or `[` wouldn't work. If you're interested in working on that, I think that'd be a good improvement. We should probably get buy in from the other maintainers first though.

In the meantime, I suggest handling the `test_foo.py::` case minimally in this PR, to fix the original issue, and open a new issue for the other suggested changes, as they go beyond #10273.